go/interpreter/lfvm/instructions.go

```diff
@@ -723,19 +723,46 @@ func opSelfdestruct(c *context) {
 		return
 	}
 
-	gasfunc := gasSelfdestruct
+	beneficiary := tosca.Address(c.stack.pop().Bytes20())
+	// Selfdestruct gas cost defined in EIP-105 (see https://eips.ethereum.org/EIPS/eip-150)
+	cost := tosca.Gas(0)
 	if c.isAtLeast(tosca.R09_Berlin) {
-		gasfunc = gasSelfdestructEIP2929
+		// as https://eips.ethereum.org/EIPS/eip-2929#selfdestruct-changes says,
+		// selfdestruct does not charge for warm access
+		if accessStatus := c.context.AccessAccount(beneficiary); accessStatus != tosca.WarmAccess {
```

Comment:
Sometimes we check for `== tosca.ColdAccess` and sometimes we check for `!= tosca.WarmAccess`. I guess this is yet another side effect of knowing that the value is boolean. 
nit: I would write `== tosca.ColdAccess`

Reply:
actually this was on purpose, the documentation of this eip says "it does not charge for warm access", hence I wrote the check in this manner to reflect it.